samples/star/src/commonMain/kotlin/com/slack/circuit/star/home/HomeScreen.kt

```diff
@@ -82,13 +88,29 @@ fun HomeContent(state: HomeScreen.State, modifier: Modifier = Modifier) {
       }
     },
   ) { paddingValues ->
+    val saveableStateHolder = rememberSaveableStateHolder()
+    val currentScreen = state.navItems[state.selectedIndex].screen
+    saveableStateHolder.SaveableStateProvider(currentScreen) {
+      val circuit = requireNotNull(LocalCircuit.current)
+      val ui = rememberUi(currentScreen, factory = circuit::ui)
+      val presenter =
+        rememberPresenter(
+          currentScreen,
+          navigator =
+            Navigator.navEventNavigator(currentScreen) { event ->
+              state.eventSink(ChildNav(event))
+            },
+          factory = circuit::presenter,
+        )
+
+      CircuitContent(
+        screen = currentScreen,
+        modifier = Modifier.padding(paddingValues),
+        presenter = presenter!!,
+        ui = ui!!,
+      )
+    }
```

Comment:
Re: loading. I think the solution here is to do something like having a map of nav items to their presenter state that is instantiated and rememberRetained above this section. 

And then have a "presenter facade" which switches state to the presenter state that is currently being shown.

This is effectively what I'm doing in my project.

Reply:
It's because we don't retain state for root back stack changes, only back stack additions (effectively).

This has been at the back of my mind for a while (and mildly annoys me every time I use Tivi).

So this change stores the tab index across app starts, but the content of that tab won't be retained if you switch between them.

Reply:
I can't find it now, but I think this might actually be as simple as making `buildCircuitContentProviders` a bit smarter, and manually retaining any dropped root records.